### DISCUSSION TITLE
How to migrate content (including media files)

### DISCUSSION THREAD

**Discussion by vijaykhollam** [Sunbird-Knowlg/Community#232] [Q&A]
We are planning to migrate content from an older website (not built on Sunbird) to a new web application that uses the Sunbird platform. As part of the migration process, I need guidance on how to handle the following:

1. **Media Files Migration**:
   - The older site uses a different cloud storage for files.
   - The new site will use Amazon S3 for media storage.
   - How can we migrate these media files effectively? Are there any specific APIs or scripts that Sunbird provides for this purpose?

2. **Migration Script or API**:
   - Do we need to write a custom script for the migration, or does Sunbird provide existing APIs or tools to facilitate this process?
   - If APIs exist, can they handle both content and media files during the migration?

---

#### Key Considerations:
- **Content Framework**: The new site will follow Sunbird's framework standards for content and metadata.
- **Storage Transition**: Media files need to be migrated from the older storage solution to the S3 bucket for the new site.
- **Automation**: We aim to automate this process as much as possible to ensure consistency and minimize manual effort.

#### Questions:
Are there specific Sunbird APIs or existing scripts for migrating content and media files from a non-Sunbird site to a Sunbird-based web application?

Any guidance, examples, or references would be greatly appreciated. Thank you!

@coolbung @snehal0904 @siddhi0719 @komalm @vinod303 @baljeetsingh

**Comment by pallakartheekreddy** [comment]
@vijaykhollam 
Thank you for your query! Could you please share the following details?

**Current Sunbird Version**: What version of the Sunbird platform are you using? This will help determine the availability of specific APIs or tools that Sunbird might provide.

**Current Cloud Provider**: Where is your data currently stored? Knowing the current cloud storage provider will help suggest the best approach for migrating to Amazon S3.

Looking forward to your response!

**Reply by vijaykhollam** [reply]
@pallakartheekreddy Thanks for quick reply. Please find details as below.

**Current Sunbird Version** : We have set up two services on server. i.e knowlge and inQuiry 
We have installed release-7.0.0 through Sunbird installer.

Repo : https://github.com/project-sunbird/sunbird-ed-installer.git
Branch : release-7.0.0

**Current Cloud Provider** : Azure Blob Storage and for new application we will be using Amazon S3.

**Current Database** : Microsoft SQL Server

**Reply by pallakartheekreddy** [reply]
@vijaykhollam Apologies, I assumed your old website was also a Sunbird instance. If it is not, you will need to write your own custom scripts for the data migration.

**Reply by vijaykhollam** [reply]
Yes, the old website is not a Sunbird instance.

**Reply by vijaykhollam** [reply]
@pallakartheekreddy Is there an API available for bulk content or course creation, such as an import feature, that we can use in the script?

**Reply by pallakartheekreddy** [reply]
@vijaykhollam No. We don't have bulk course creation APIs.

**Reply by vijaykhollam** [reply]
Thanks @pallakartheekreddy
